cub/cub/detail/choose_offset.cuh

```diff
@@ -65,6 +65,22 @@ struct ChooseOffsetT
     typename std::conditional<sizeof(NumItemsT) <= 4, std::uint32_t, unsigned long long>::type;
 };
 
+/**
+ * PromoteSmallOffsetT checks NumItemsT, the type of the num_items parameter, and
+ * promotes any integral type smaller than 32 bits to a signed 32-bit integer type.
+ */
+template <typename NumItemsT>
+struct PromoteSmallOffsetT
+{
+  // NumItemsT must be an integral type (but not bool).
+  static_assert(std::is_integral<NumItemsT>::value
```

Comment:
important: let's use `::cuda::std` in new code (here and below). This is needed for NVRTC support. I don't think that this header will break NVRTC, since it's related to device-scope algorithms, but it's a good practice to keep in mind. Also, we need to update the developer overview to state that.

Reply:
Good point, thanks. Switched to `::cuda::std`. Planning to do the same for `ChooseOffsetT` in a separate PR.